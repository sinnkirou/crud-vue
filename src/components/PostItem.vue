<template>
  <div class="post" key="{{`PostItem${post.id}`}}">
    <div v-if="!editable">
      <h2 class="post_title">{{post.title}}</h2>
      <p class="post_message">{{post.message}}</p>
      <p class="post_id">{{`id: ${post.id}`}}</p>
      <div class="control_buttons">
        <button class="edit" type="button" @click.prevent="toggleEditable">{{'Edit'}}</button>
        <button class="delete" type="button" @click.prevent="deletePost(post.id)">{{'Delete'}}</button>
      </div>
    </div>
    <EditablePost v-else-if="editable" :post="post" v-on:toggle-Editable="toggleEditable"></EditablePost>
  </div>
</template>

<script>
export default {
  name: "PostItem",
  props: {
    post: Object
  },
  components: {
    EditablePost: () => import("./EditablePost")
  },
  data: () => ({
    editable: false
  }),
  inject: ["deletePost"],
  methods: {
    toggleEditable: function() {
      this.editable = !this.editable;
    }
  }
};
</script>