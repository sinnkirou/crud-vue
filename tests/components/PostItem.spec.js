import { shallowMount } from "@vue/test-utils";
import PostItem from "@/components/PostItem.vue";
import { localVue, store, posts } from "../mockStore";
import sinon from "sinon";

describe("PostItem.vue", () => {
  let wrapper;
  let vm;
  let sandbox;
  const post = posts[0];

  beforeEach(() => {
    wrapper = shallowMount(PostItem, {
      propsData: { post },
      localVue,
      store
    });
    vm = wrapper.vm;
    sandbox = sinon.createSandbox();
  });

  afterEach(() => {
    wrapper.destroy();
    sandbox.restore();
  });

  it("should render", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.vm.post).toEqual(post);
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should toggle editable", () => {
    expect(vm.editable).toEqual(false);
    var editBtn = wrapper.find("button.edit");
    editBtn.trigger("click");
    expect(vm.editable).toEqual(true);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should toggle delete", () => {
    const deleteSpy = sandbox.spy(vm, "DELETE_POST");

    var deleteBtn = wrapper.find("button.delete");
    deleteBtn.trigger("click");
    expect(deleteSpy.calledOnce).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
