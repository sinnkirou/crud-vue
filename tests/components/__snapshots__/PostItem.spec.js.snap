// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem.vue has the expected html structure 1`] = `
<div
  class="post"
>
  <div>
    <h2
      class="post_title"
    >
      title
    </h2>
     
    <p
      class="post_message"
    >
      message
    </p>
     
    <p
      class="post_id"
    >
      id: 1
    </p>
     
    <div
      class="control_buttons"
    >
      <button
        class="edit"
        type="button"
      >
        Edit
      </button>
       
      <button
        class="delete"
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;

exports[`PostItem.vue should toggle delete 1`] = `
<div
  class="post"
>
  <div>
    <h2
      class="post_title"
    >
      title
    </h2>
     
    <p
      class="post_message"
    >
      message
    </p>
     
    <p
      class="post_id"
    >
      id: 1
    </p>
     
    <div
      class="control_buttons"
    >
      <button
        class="edit"
        type="button"
      >
        Edit
      </button>
       
      <button
        class="delete"
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;

exports[`PostItem.vue should toggle editable 1`] = `
<div
  class="post"
>
  <!---->
</div>
`;
